src/dlgPackageExporter.cpp

```diff
@@ -286,7 +286,7 @@ void dlgPackageExporter::slot_export_package()
             zip_error_to_str(buf, sizeof(buf), err, errno);
             //FIXME: report error to userqDebug()<<"zip source error"<<fullName<<fname<<buf;
         }
-        err = zip_add(archive, fname.toStdString().c_str(), s);
+        err = zip_file_add(archive, fname.toStdString().c_str(), s, ZIP_FL_OVERWRITE);
```

Comment:
Um, I was testing this and found it is still not enough to fix things - it does not handle sub-directories and will silently fail should they be present, (with an "unknown error" when the `zip_close(archive)` call is made {on line `297`}) when all the work is done and finds that it is trying to add a directory and not a file with a `zip_file_add` call! :slightly_frowning_face: 

To cure that I suggest you need to make two passes through `contents` and in the first pass identify the entries that are sub-directories, use `zip_dir_add` to put them into the archive then make a second pass and identify the remaining files to insert/replace with `zip_file_add`. You will want to consider what other entries might appear, e.g. symbolic links, junction {or other re-parse} points} and avoid them - which can be done by basically, on line 275 changing:

`
QStringList contents = dir.entryList();
`

to:
`
QStringList contents = dir.entryList(QDir::AllDirs|QDir::Files|QDir::NoSymLinks|QDir::NoDotAndDotDot);
`

This will filter out the `.` and `..` entries so that lines 278 to 280 then become redundant.

For each entry you will want a `QFileInfo` instance for it - so it may be more efficient to use `QDir::entryInfoList(`...`)` rather than `QDir::entryList(`...`)` and to iterate through that - then you can use `QFileInfo::isDir()` to identify the directories and `QFileInfo::isFile()` to spot the files.  Furthermore given that there could be nested sub-directories you may want to gather them all up and then create them, at the end of the first path, in order of length (shortest first) in the archive so that you do not have a sub-directory before it's parent!

Addressing this will *finally* mean that sub-directories can be included in the archive as has previously seemed to have been promised by the exporter system - note that the companion code to *read* sub-directories in mudlet modules has been around for a year or more (see https://github.com/Mudlet/Mudlet/commit/fcb8d64f64469c2a4cd9538eeff3ae14a9288c0b).